packages/arb-node-core/challenge/generateProofCases.go

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright 2019-2020, Offchain Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package challenge
+
+import (
+	"fmt"
+	"math/big"
+
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/pkg/errors"
+
+	"github.com/offchainlabs/arbitrum/packages/arb-avm-cpp/cmachine"
+	"github.com/offchainlabs/arbitrum/packages/arb-util/core"
+	"github.com/offchainlabs/arbitrum/packages/arb-util/inbox"
+	"github.com/offchainlabs/arbitrum/packages/arb-util/machine"
+)
+
+type ExecutionCutJSON struct {
+	GasUsed      uint64
+	InboxDelta   ethcommon.Hash
+	MachineState ethcommon.Hash
+	SendAcc      ethcommon.Hash
+	SendCount    *hexutil.Big
+	LogAcc       ethcommon.Hash
+	LogCount     *hexutil.Big
+}
+
+type ProofData struct {
+	BeforeCut   ExecutionCutJSON
+	AfterCut    ExecutionCutJSON
+	Proof       hexutil.Bytes
+	BufferProof hexutil.Bytes
+}
+
+func GenerateProofCases(contract string, fuzzTarget bool) ([]*ProofData, error) {
```

Comment:
With the other two uses of `fuzzTarget` eliminated, we can just pass `maxSteps` in here as an arg